drivers/disk/ftl_dhara.c

```diff
@@ -0,0 +1,424 @@
+/*
+ * Copyright (c) 2025 Endress+Hauser GmbH+Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_ftl_dhara
+
+#include <dhara/map.h>
+#include <dhara/nand.h>
+#include <zephyr/drivers/disk.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/drivers/flash/nand_flash_api_ex.h>
+#include <zephyr/storage/flash_map.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(disk_ftl_dhara, CONFIG_DISK_FTL_LOG_LEVEL);
+
+struct disk_ftl_data {
+	struct k_sem lock;
+	struct disk_info info;
+	uint8_t partition_id;
+	off_t partition_offset;
+	size_t page_size;
+	size_t block_size;
+	size_t partition_size;
+	uint8_t *page_buffer;
+	struct dhara_map dhara_map;
+	struct dhara_nand dhara_nand;
+	uint8_t *dhara_buffer;
+	uint8_t dhara_gc_ratio;
+};
+
+int dhara_nand_is_bad(const struct dhara_nand *n, dhara_block_t b)
+{
+	struct disk_ftl_data *ctx = CONTAINER_OF(n, struct disk_ftl_data, dhara_nand);
+	int is_bad = 1;
+	int ret;
+
+	ret = flash_ex_op(ctx->info.dev, NAND_FLASH_IS_BAD_BLOCK, (uintptr_t)&b, &is_bad);
+	if (ret == -ENOTSUP) {
+		LOG_INF("Checking bad block is not supported");
```

Comment:
LOG_DBG as this is only needed on testing and evaluation of user system setup. Otherwise you will have a lot of these.